### DISCUSSION TITLE
How I get the directory name up level?

### DISCUSSION THREAD

**Discussion by weyderfs** [gruntwork-io/knowledge-base#625] [How do I] [answered]
How can I get the name of the directory up the level where my `terragrunt.hcl` is?

## Example

- project-name/
   - sqs/
      - `terragrunt.hcl`

In this context bellow I need to get `project-name`, I've tried the followings functions without success:

* `${basename(${get_terragrunt_dir()}../)}`
* `basename(${get_terragrunt_dir()}../)`
* `${basename(get_terragrunt_dir())../}`
* `"${run_cmd("ls -l ../../ | grep '^d' | awk -F' ' '{print $11}'')}"` (In this context the problem is about the double quotes 😅)
* `dirname("../../")`
* `basename("../../")`

Maybe using the `relative path + trimprefix`, but I dont know how to combine and use them 😞 .

---

<ins datetime="2023-01-02T19:12:11Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109756">Tracked in ticket #109756</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
depends on how the repository is organized - if `project-name` is also root of git repository, can be used `get_repo_root()` function:

```
locals {
  root_dir = run_cmd("echo", "${basename(get_repo_root())}")
}

```
https://terragrunt.gruntwork.io/docs/reference/built-in-functions/#get_repo_root

If this is not the case, and is only 2 levels of hierarchy, can be combined `get_terragrunt_dir` and `dirname`:

```
locals {
  path = "${get_terragrunt_dir()}/../.."
  root_dir = "${dirname(local.path)}"
}
```

Since concatenation is not working, additional strings can be added through variables

https://terragrunt.gruntwork.io/docs/reference/built-in-functions/#get_terragrunt_dir

**Reply by weyderfs** [reply]
@denis256 thanks for help, In my case I use the second example because `project-name` not is the _root_. but now is showing the _ full path_ and not only de `project-name` folder.
I'm trying to adapt your example with _trimprefix_ or other method, but if you know something will help me.

My adaptation:
```hcl
locals {
  path = "${path_relative_to_include()}/../.."
  root_dir = run_cmd("echo", "${dirname(local.path)}")
  name = trimprefix("${local.root_dir}", "../..")
}
terraform {
  source = [...]
}

input = {
   name = local.name
   [...]
}

include {
  path = find_in_parent_folders()
}
```

The output:

```sh
Unsupported operator; Bitwise operators are not supported. Did you mean boolean OR ("||")?, and 7 other diagnostic(s)
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

**Comment by Fabianoshz** [comment, marked as answer]
Try this:


```
locals {
  parent = "${get_terragrunt_dir()}/../"
  name = basename(dirname(local.parent))
}

input = {
   name = local.name
}
```

**Reply by weyderfs** [reply]
I will do more tests, but at first your example solve to me.
